tests/scripts/thread-cert/border_router/test_multi_ail.py

```diff
@@ -38,7 +38,126 @@
+            'allowlist': [],
+            'is_otbr': True,
+            'version': '1.3',
+        },
+        BR1_0: {
+            'name': 'BR1_1',
+            'backbone_network_id': 0,
+            'allowlist': [BR1_1],
+            'is_otbr': True,
+            'version': '1.3',
+        },
+        BR1_1: {
+            'name': 'BR1_2',
+            'backbone_network_id': 2,
+            'allowlist': [BR1_0],
+            'is_otbr': True,
+            'version': '1.3',
+        },
+        BR2_0: {
+            'name': 'BR2_0',
+            'backbone_network_id': 0,
+            'allowlist': [BR2_1],
+            'is_otbr': True,
+            'version': '1.3',
+        },
+        BR2_1: {
+            'name': 'BR2_1',
+            'backbone_network_id': 1,
+            'allowlist': [BR2_0, BR2_2],
+            'is_otbr': True,
+            'version': '1.3',
+        },
+        BR2_2: {
+            'name': 'BR2_2',
+            'backbone_network_id': 1,
+            'allowlist': [BR2_1, BR2_3],
+            'is_otbr': True,
+            'version': '1.3',
+        },
+        BR2_3: {
+            'name': 'BR2_4',
+            'backbone_network_id': 2,
+            'allowlist': [BR2_2],
+            'is_otbr': True,
+            'version': '1.3',
+        }
+    }
+
+    def test(self):
+        """ This test verify "br peers" and "br routers" commands work as expected.
+        """
+        br0_0: OtbrNode = self.nodes[self.BR0_0]
+        br1_0: OtbrNode = self.nodes[self.BR1_0]
+        br1_1: OtbrNode = self.nodes[self.BR1_1]
+        br2_0: OtbrNode = self.nodes[self.BR2_0]
+        br2_1: OtbrNode = self.nodes[self.BR2_1]
+        br2_2: OtbrNode = self.nodes[self.BR2_2]
+        br2_3: OtbrNode = self.nodes[self.BR2_3]
+
+        border_routers = [br0_0, br1_0, br1_1, br2_0, br2_1, br2_2, br2_3]
+
+        for br in border_routers:
+            br.start()
+
+        # sleep some time more to ensure every BR gets the network data, otherwise the test can fail sometimes.
+        self.simulator.go(config.BORDER_ROUTER_STARTUP_DELAY + 3)
+
+        # ---------------------------------------
+        # br peers
+        _, br1_0_rloc16, br1_1_rloc16, br2_0_rloc16, br2_1_rloc16, br2_2_rloc16, br2_3_rloc16 = [
+            br.get_addr16() for br in border_routers
+        ]
+
+        self.assertCountEqual(br0_0.get_br_peers_rloc16s(), [])
+        self.assertCountEqual(br1_0.get_br_peers_rloc16s(), [br1_1_rloc16])
+        self.assertCountEqual(br1_1.get_br_peers_rloc16s(), [br1_0_rloc16])
+        self.assertCountEqual(br2_0.get_br_peers_rloc16s(), [br2_1_rloc16, br2_2_rloc16, br2_3_rloc16])
+        self.assertCountEqual(br2_1.get_br_peers_rloc16s(), [br2_0_rloc16, br2_2_rloc16, br2_3_rloc16])
+        self.assertCountEqual(br2_2.get_br_peers_rloc16s(), [br2_0_rloc16, br2_1_rloc16, br2_3_rloc16])
+        self.assertCountEqual(br2_3.get_br_peers_rloc16s(), [br2_0_rloc16, br2_1_rloc16, br2_2_rloc16])
```

Comment:
It's unclear to me what the primary objective of this test is.

Are we validating the `br peers` or `otBorderRoutingGetNextPeerBrEntry()` APIs? These seem to be already covered in existing tests, like `test-503-peer-tbr-discovery.py`. While that test defines BRs on multiple AILs, the presence of multiple AILs isn't directly relevant to the functionality of the `br peers` APIs. I don't see any checks in this new test that are specifically tied to the multi-AIL aspect of the network topology.

Reply:
Sounds reasonable!  
My original thought for this test function is that I want to test `br peers` and `br routers` (along with the OT API)  work perfectly under the complex multi-ail, multi-partition topology from the system wide view.  (Maybe I need to add some more test case, e.g. device offline, device join network not at the same time, I but don't know how to accurately define them yet 😕) So in this PR, I just add some checks as more as possible. Any thoughts?

Reply:
I would suggest we consider adding this PR when we have some additional test steps that leverage the multi-AIL aspect of the topology under test.

> My original thought for this test function is that I want to test `br peers` and `br routers` (along with the OT API) work perfectly under the complex multi-ail, multi-partition topology from the system wide view.

The OT `RoutingManager` which provides `br peers` learns this by processing received RA messages. The multi-AIL aspect is not  relevant for this. This test case isn't really validating anything new within the OT stack itself, but rather it's verifying that the simulated test topology is correctly set up, i.e., validating that RAs from one AIL are not inadvertently forwarded to another AIL.  It is fine to validate that test topology is functioning correctly, but on its own it cannot be the full test. It should be followed by some additional test steps that utilize the multi-AIL aspect and validate some function within OT related to multi-AIL.

Reply:
Thanks Abtin, sound good! I will close this PR now.

Reply:
pushed a new PR https://github.com/openthread/openthread/pull/10637 for updating the multi_ail test case